From 4e83d3c855c8cf1e0ac1cc782bcd0b68dd6936bd Mon Sep 17 00:00:00 2001
From: Chenghua Xu <xuchenghua@loongson.cn>
Date: Thu, 17 May 2018 15:36:32 +0800
Subject: [PATCH 13/15] Disable gss[bhwd] and gsl[bhwd] on o32 abi.

    gcc/config/mips/
    * mips.md: Added TARGET_64BIT in index address load/store.
---
 gcc/config/mips/mips.md | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gcc/config/mips/mips.md b/gcc/config/mips/mips.md
index 5c6278a375e..1d9534880af 100644
--- a/gcc/config/mips/mips.md
+++ b/gcc/config/mips/mips.md
@@ -4892,7 +4892,7 @@
         (mem:GPR
                 (plus:P (match_operand:P 1 "register_operand" "d")
                         (match_operand:P 2 "register_operand" "d"))))]
-  "TARGET_LOONGSON_3A"
+  "TARGET_LOONGSON_3A && TARGET_64BIT"
   "<GPR:gsloadx>\t%0,0(%1,%2)"
   [(set_attr "type" "load")
    (set_attr "mode" "<GPR:MODE>")])
@@ -4901,7 +4901,7 @@
   [(set (mem:GPR (plus:P (match_operand:P 1 "register_operand" "d")
                          (match_operand:P 2 "register_operand" "d")))
         (match_operand:GPR 0 "register_operand" "d"))]
-  "TARGET_LOONGSON_3A"
+  "TARGET_LOONGSON_3A && TARGET_64BIT"
   "<GPR:gsstorex>\t%0,0(%1,%2)"
   [(set_attr "type" "store")
    (set_attr "mode" "<GPR:MODE>")])
@@ -4913,7 +4913,7 @@
           (mem:SHORT
             (plus:P (match_operand:P 1 "register_operand" "d")
                     (match_operand:P 2 "register_operand" "d")))))]
-  "TARGET_LOONGSON_3A"
+  "TARGET_LOONGSON_3A && TARGET_64BIT"
   "<SHORT:gsloadx>\t%0,0(%1,%2)"
   [(set_attr "type" "load")
    (set_attr "mode" "<GPR:MODE>")])
@@ -4922,7 +4922,7 @@
   [(set (mem:SHORT (plus:P (match_operand:P 1 "register_operand" "d")
                            (match_operand:P 2 "register_operand" "d")))
         (match_operand:SHORT 0 "register_operand" "d"))]
-  "TARGET_LOONGSON_3A"
+  "TARGET_LOONGSON_3A && TARGET_64BIT"
   "<SHORT:gsstorex>\t%0,0(%1,%2)"
   [(set_attr "type" "store")
    (set_attr "mode" "SI")])
-- 
2.16.1

